src/khoj/routers/api.py

```diff
@@ -746,11 +752,11 @@ def edit_job(
     user_timezone = pytz.timezone(timezone)
     trigger = CronTrigger.from_crontab(crontime, user_timezone)
     if automation.trigger != trigger:
-        automation.reschedule(trigger=trigger)
+        await sync_to_async(automation.reschedule)(trigger=trigger)
 
     # Collate info about the updated user automation
-    automation = AutomationAdapters.get_automation(user, automation.id)
-    automation_info = AutomationAdapters.get_automation_metadata(user, automation)
+    automation = await AutomationAdapters.aget_automation(user, automation.id)
+    automation_info = await sync_to_async(AutomationAdapters.get_automation_metadata)(user, automation)
```

Comment:
In the `get_automation_metadata` function, we had some explicit regex operations to remove the `/automated_task` prefix from `query_to_run`. Let's remove that as it's not required anymore.